recipe/0001-drop-pyarrow-version-numbers.patch

```diff
@@ -1,61 +1,36 @@
-From 8e0daac205f909e73017d4ae3fc6f2a382378859 Mon Sep 17 00:00:00 2001
-From: Andrew Achkar <hajapy@users.noreply.github.com>
-Date: Sat, 30 Apr 2022 10:16:22 -0400
-Subject: [PATCH] drop pyarrow version numbers
-
----
- pyproject.toml | 2 +-
- setup.cfg      | 2 +-
- setup.py       | 8 ++++----
- 3 files changed, 6 insertions(+), 6 deletions(-)
-
-diff --git a/pyproject.toml b/pyproject.toml
-index ad1f850..16bc0b3 100644
---- a/pyproject.toml
-+++ b/pyproject.toml
-@@ -6,7 +6,7 @@ requires = [
-     "wheel",
-     "cython",
-     # Must be kept in sync with the `setup_requirements` in `setup.cfg`
--    "pyarrow>=6.0.0,<6.1.0",
-+    "pyarrow",
- ]
-
- [tool.cibuildwheel]
-diff --git a/setup.cfg b/setup.cfg
-index 3d906e0..3180e1b 100644
---- a/setup.cfg
-+++ b/setup.cfg
-@@ -89,6 +89,6 @@ development =
-     pytzdata
- pandas =
-     pandas>=1.0.0,<1.5.0
--    pyarrow>=6.0.0,<6.1.0
-+    pyarrow
- secure-local-storage =
-     keyring!=16.1.0,<24.0.0
 diff --git a/setup.py b/setup.py
-index 9244cab..3b490b7 100644
+index d4ccdb7..dc65b49 100644
 --- a/setup.py
 +++ b/setup.py
-@@ -69,14 +69,14 @@ if _ABLE_TO_COMPILE_EXTENSIONS:
+@@ -69,23 +69,19 @@ if _ABLE_TO_COMPILE_EXTENSIONS:
+ 
+         # list of libraries that will be bundled with python connector,
          # this list should be carefully examined when pyarrow lib is
-         # upgraded
+-        # upgraded
++        # upgraded. For conda, we don't vendor at all
          arrow_libs_to_copy = {
--            "linux": ["libarrow.so.600", "libarrow_python.so.600"],
--            "darwin": ["libarrow.600.dylib", "libarrow_python.600.dylib"],
-+            "linux": ["libarrow.so", "libarrow_python.so"],
-+            "darwin": ["libarrow.dylib", "libarrow_python.dylib"],
-             "win32": ["arrow.dll", "arrow_python.dll"],
+-            "linux": ["libarrow.so.800", "libarrow_python.so.800", "libparquet.so.800"],
+-            "darwin": [
+-                "libarrow.800.dylib",
+-                "libarrow_python.800.dylib",
+-                "libparquet.800.dylib",
+-            ],
+-            "win32": ["arrow.dll", "arrow_python.dll", "parquet.dll"],
++            "linux": [],
++            "darwin": [],
++            "win32": [],
```

Comment:
I need to explain what is happening here. These arrays were defining a set of dynamic libraries _to be vendored into the pyarrow package itself._ You can check our older builds of snowflake-connector-python to see that they are indeed in there. With this new patch I am removing those libraries altogether in favor of our arrow-cpp dependency.

I have personally spoken to the Snowflake developers who agree this is an entirely legitimate approach. One of the reasons they do this is actually because they only need a small fraction of the arrow-cpp library set, so doing this allows users of the pip package to build more compact environments; e.g., for AWS Lambdas.

Reply:
Agree, they shouldn't be vendored into the package. Using the unversioned solinks below is useful for future package builds, as long as the patch applies. Same for pyarray in the pyproject.toml, as long as we have a constraint in meta.yaml.